libasn1compiler/asn1c_misc.c

```diff
@@ -61,8 +61,18 @@ asn1c_make_identifier(enum ami_flags_e flags, asn1p_expr_t *expr, ...) {
 		/*
 		 * Estimate the necessary storage size
 		 */
-		if(expr->Identifier == NULL)
-			return "Member";
+		if(expr->Identifier == NULL) {
+			if(storage_size < 20) {
+				storage = realloc(storage, 20);  /* reserve space for 'Memberxxx' */
```

Comment:
Since you're not reusing the contents of the buffer, I would recommend free+malloc here instead. It also avoids the memory leak that would happen if realloc returned NULL.